migrator/migrations/m_170_to_m_171_move_scope_to_collection_in_report_configurations/migration.go

```diff
@@ -0,0 +1,247 @@
+		index = index + 1
+	}
+	return collections, true
+}
+
+func createCollectionsToEmbedFromScope(scope *storage.SimpleAccessScope) ([]*storage.ResourceCollection, bool) {
+	collectionsToEmbed := make([]*storage.ResourceCollection, 0)
+
+	index := 0
+	if includedClusters := scope.GetRules().GetIncludedClusters(); len(includedClusters) > 0 {
+		ruleVals := make([]*storage.RuleValue, 0, len(includedClusters))
+		for _, cluster := range includedClusters {
+			ruleVals = append(ruleVals, &storage.RuleValue{
+				Value:     cluster,
+				MatchType: storage.MatchType_EXACT,
+			})
+		}
+		col := buildEmbeddedCollection(scope.GetName(), index, []*storage.SelectorRule{
+			{
+				FieldName: search.Cluster.String(),
+				Operator:  storage.BooleanOperator_OR,
+				Values:    ruleVals,
+			},
+		})
+		collectionsToEmbed = append(collectionsToEmbed, col)
+		index = index + 1
+	}
+
+	if includedNamespaces := scope.GetRules().GetIncludedNamespaces(); len(includedNamespaces) > 0 {
+		for _, namespace := range includedNamespaces {
+			col := buildEmbeddedCollection(scope.GetName(), index, []*storage.SelectorRule{
+				{
+					FieldName: search.Cluster.String(),
+					Operator:  storage.BooleanOperator_OR,
+					Values: []*storage.RuleValue{
+						{
+							Value:     namespace.GetClusterName(),
+							MatchType: storage.MatchType_EXACT,
+						},
+					},
+				},
+				{
+					FieldName: search.Namespace.String(),
+					Operator:  storage.BooleanOperator_OR,
+					Values: []*storage.RuleValue{
+						{
+							Value:     namespace.GetNamespaceName(),
+							MatchType: storage.MatchType_EXACT,
+						},
+					},
+				},
+			})
+			collectionsToEmbed = append(collectionsToEmbed, col)
+			index = index + 1
+		}
+	}
+
+	if clusterLabelSelectors := scope.GetRules().GetClusterLabelSelectors(); len(clusterLabelSelectors) > 0 {
+		labelCollections, success := labelSelectorsToCollections(scope.GetName(), index, clusterLabelSelectors, search.ClusterLabel.String())
+		if !success {
+			return nil, false
+		}
+		index = index + len(labelCollections)
+		collectionsToEmbed = append(collectionsToEmbed, labelCollections...)
+	}
+
+	if namespaceLabelSelectors := scope.GetRules().GetNamespaceLabelSelectors(); len(namespaceLabelSelectors) > 0 {
+		labelCollections, success := labelSelectorsToCollections(scope.GetName(), index, namespaceLabelSelectors, search.NamespaceLabel.String())
+		if !success {
+			return nil, false
+		}
+		collectionsToEmbed = append(collectionsToEmbed, labelCollections...)
+	}
+	return collectionsToEmbed, true
+}
+
+func moveScopeIDToCollectionIDInReports(db *pgxpool.Pool) error {
+	ctx := context.Background()
+	reportConfigStore := reportConfigurationPostgres.New(db)
+	accessScopeStore := accessScopePostgres.New(db)
+	collectionStore := collectionPostgres.New(db)
+
+	reportConfigsToUpsert := make([]*storage.ReportConfiguration, 0, batchSize)
+	err := reportConfigStore.Walk(ctx, func(reportConfig *storage.ReportConfiguration) error {
```

Comment:
> a. No collections exist in database when this migration runs.
> b. Simple access scope ids are UUIDs
> 
> Given the above premise, you do not actually need update report configuration with new UUID. You can create root collections for scopes in reports using exact same ID.

With above, you can split the migration in two steps.
1. Collect scope IDs from report configurations.
2. Create Collections for scopes from (1) with same ID for root collection.

Reply:
Thanks, that simplified the implementation a lot.